internal/infrastructure/middleware/enforce_identity.go

```diff
@@ -13,6 +14,15 @@ import (
 	identity "github.com/redhatinsights/platform-go-middlewares/v2/identity"
 )
 
+// https://github.com/RedHatInsights/identity-schemas/blob/main/3scale/schema.json#L21
```

Comment:
Line 21 points to `auth_type` and not `type`. Also `type` there has only 3 values - `User`, `ServiceAccount`, `System`. 

The: `X509` and `Associate` are bit weird. It's coming from  in https://github.com/RedHatInsights/platform-go-middlewares/blob/master/identity/identity.go#L101 right?  I also saw them used as type in https://github.com/RedHatInsights/platform-go-middlewares/blob/master/identity/identity_suite_test.go so I guess we should rather work with them - so OK.

Reply:
good catch! I have updated the link! thanks!

Reply:
you are right; `X509` and `Associate` are comming from the link you referenced; I added them as I have seen in the json schema definition at: https://github.com/RedHatInsights/identity-schemas/blob/main/turnpike/schema.json just for considering all the potential values and reflect the mapping in the log, and avoid `unknown` for them.